libbeat/cmd/instance/beat.go

```diff
@@ -593,6 +597,16 @@ func (b *Beat) configure(settings Settings) error {
 		imFactory = idxmgmt.MakeDefaultSupport(settings.ILM)
 	}
 	b.index, err = imFactory(nil, b.Beat.Info, b.RawConfig)
+	if err != nil {
+		return err
+	}
+
+	processingFactory := settings.Processing
```

Comment:
Why is the `ProcessorFactory` defined on the root cmd level, but the `ProcessingConfig` on the pipeline level? Would it make sense to move the processorFactory also to the processing config level?

Reply:
Beats do not directly publish methods to the queue in the publisher pipeline, but do so via a `Client`.
The pipeline including global processors and fields settings, queue, and outputs is created on startup. There are no go-routines collecting data yet.
Go-routines are supposed to use `Connect`, so to connect to a publisher pipeline. This returns a `beat.Client` (Client instances are not guaranteed to be thread safe). Each client/go-routine is allowed to configure local processors/fields/tags, which gets merged with the global settings. The factory is the global entity, loading, checking, and preparing global configurations on startup. The ProcessingConfig specifies the per go-routine local processing, which might be eventually established due to the Beat modules initialization or much later via autodiscovery.